### PR TITLE
Fariba

### PR DESCRIPTION
**Pull request by faribas**
Updated PD- Ordinal with xgxr functions.

### CODE REVIEW

File: Rmarkdown/Adverse_Events.Rmd [hunk 1]
```diff
@@ -46,6 +38,7 @@ my.data <- read.csv("../Data/AE_xgx.csv")
 my.data$DOSE_label <- paste(my.data$Dose,"mg")
```
Comment (margoal1) [inline Rmarkdown/Adverse_Events.Rmd]: maybe should rename "my.data" as "ae_data"

Comment (margoal1) [inline Rmarkdown/Adverse_Events.Rmd]: What are the required variable  names for this dataset? Adding a block of variable name definitions like in the other pages would help the user to understand what variables are needed.

Comment (margoal1) [reply Rmarkdown/Adverse_Events.Rmd]: Include DOSE_label & AE_binary definitions in this block of definitions

File: Rmarkdown/Multiple_Ascending_Dose_PD_ordinal.Rmd [hunk 2]
```diff
@@ -16,92 +17,98 @@ Ordinal data can be thought of as categorical data that has a natural order. For
...
-  
-  return(annotateStatus)
-}
-
-```
-
 ## Load Dataset
 
 ```{r, warning=FALSE, message=FALSE}
 my.data <- read.csv("../Data/Multiple_Ascending_Dose_Dataset2.csv")
```
Comment (margoal1) [inline Rmarkdown/Multiple_Ascending_Dose_PD_ordinal.Rmd]: rename as pkpd_data

File: Rmarkdown/Multiple_Ascending_Dose_PD_ordinal.Rmd [hunk 3]
```diff
@@ -16,92 +17,98 @@ Ordinal data can be thought of as categorical data that has a natural order. For
...
+         NOMTIME = NOMTIME,#NOMINAL TIME column name
+         EVID    = EVID   ,#EVENT ID, >=1 is dose, otherwise measurement
+         PROFDAY = PROFDAY,#PROFILE DAY day associated with profile, e.g. day of dose administration
+         LIDV    = LIDV,   #DEPENDENT VARIABLE column name
+         CENS    = CENS,   #CENSORING column name
+         CMT     = CMT,    #COMPARTMENT column
+         DOSE    = DOSE,   #DOSE column here (numeric value)
+         TRTACT  = TRTACT, #DOSE REGIMEN column here (character, with units),
+         )
 my.data$Severity <- my.data$LIDV
```
Comment (margoal1) [inline Rmarkdown/Multiple_Ascending_Dose_PD_ordinal.Rmd]: can include these lines in mutate above

File: Rmarkdown/Multiple_Ascending_Dose_PD_ordinal.Rmd [hunk 4]
```diff
@@ -160,84 +161,61 @@ gg
...
 ```{r, cache = TRUE, fig.width = 5, fig.height = 4, warning = FALSE, message= FALSE}
-data_to_plot <- my.data[my.data$CMT==5,]
-data_to_plot$DAY_label <- paste("Day", data_to_plot$PROFDAY)
-data_to_plot$DAY_label[data_to_plot$DAY_label=="Day 0"] = "Baseline"
-data_to_plot <- data_to_plot[data_to_plot$DAY_label%in%c("Day 5"),]
-
-# data_to_plot$DAY_label <- factor(data_to_plot$DAY_label, levels = rev(sort(unique(data_to_plot$DAY_label))))
-
-data_to_plot$Response <- factor(data_to_plot$Severity_label, levels = rev(levels(data_to_plot$Severity_label)))
+pd_data_d5 <- pd_data[pd_data$DAY_label%in%c("Day 5"),]
```
Comment (margoal1) [inline Rmarkdown/Multiple_Ascending_Dose_PD_ordinal.Rmd]: use %>%

File: Rmarkdown/Multiple_Ascending_Dose_PD_ordinal.Rmd [hunk 5]
```diff
@@ -160,84 +161,61 @@ gg
...
 ### Percent of subjects by response category against dose, faceted by day
 
 ```{r, cache = TRUE, fig.width = 8, fig.height = 3, warning = FALSE, message= FALSE}
-data_to_plot <- my.data[my.data$CMT==5,]
-data_to_plot$DAY_label <- paste("Day", data_to_plot$PROFDAY)
-data_to_plot$DAY_label[data_to_plot$DAY_label=="Day 0"] = "Baseline"
-data_to_plot <- data_to_plot[data_to_plot$DAY_label%in%c("Day 1","Day 3","Day 5"),]
-
-# data_to_plot$DAY_label <- factor(data_to_plot$DAY_label, levels = rev(sort(unique(data_to_plot$DAY_label))))
+pd_data_d135 <- pd_data[pd_data$DAY_label%in%c("Day 1","Day 3","Day 5"),]
```
Comment (margoal1) [inline Rmarkdown/Multiple_Ascending_Dose_PD_ordinal.Rmd]: use %>%

File: Rmarkdown/Adverse_Events.Rmd [hunk 6]
```diff
@@ -167,22 +178,22 @@ The plot below shows time vs AUC from a pop PK model at each day. The colored do
 
 ```{r, cache = TRUE, fig.width = 8, fig.height = 4, warning = FALSE, message= FALSE}
 data_to_plot <- my.data[!is.na(my.data$AETOXGRDN), ]
```
Comment (margoal1) [inline Rmarkdown/Adverse_Events.Rmd]: use %>%

File: Rmarkdown/Adverse_Events.Rmd (hunk 1 above)
Comment (faribas) [reply Rmarkdown/Adverse_Events.Rmd]: I am going to keep it my.data to be consistent with other pages

Comment (faribas) [reply Rmarkdown/Adverse_Events.Rmd]: I did it, we can take care of consistency later...

File: Rmarkdown/Oncology_Efficacy_Plots.Rmd [hunk 7]
```diff
@@ -254,29 +304,31 @@ When two drugs are combined (ABC123 + DEF456), it can be a useful first step to
 gg <- ggplot()
 gg <- gg + geom_point(data=my.data.monotherapy,aes(x=DOSE_ABC,y=BPCHG, color="ABC123"))
 gg <- gg + geom_point(data=my.data.combo,aes(x=DOSE_ABC,y=BPCHG, 
-                                          color = factor(DOSE_DEF, labels = c("ABC123 + 300 mg DEF456", "ABC123 + 400 mg DEF456"))))
+                                          color = factor(DOSE_DEF, labels = c("ABC123 + 400 mg DEF456", "ABC123 + 300 mg DEF456"))))
```
Comment (margoal1) [inline Rmarkdown/Oncology_Efficacy_Plots.Rmd]: should we add DOSE_DEF_label and DOSE_ABC_label definitions to the top of the page?

File: Rmarkdown/Oncology_Efficacy_Plots.Rmd [hunk 8]
```diff
@@ -254,29 +304,31 @@ When two drugs are combined (ABC123 + DEF456), it can be a useful first step to
...
 gg <- gg + theme(text = element_text(size=15))
+gg <- gg + xgx_annotate_status(status)
 gg
 ```
 
 
 ```{r, cache = TRUE, fig.width = 10, fig.height = 5, warning = FALSE, message= FALSE} 
 gg <- ggplot(data=my.data.combo,aes(x=DOSE_DEF,y=BPCHG))
-gg <- gg + geom_point(aes( color= factor(DOSE_ABC,labels=c("+ 20 mg ABC123",
+gg <- gg + geom_point(aes( color= factor(DOSE_ABC,labels=c("+ 60 mg ABC123",
```
Comment (margoal1) [inline Rmarkdown/Oncology_Efficacy_Plots.Rmd]: same as above

File: Rmarkdown/Oncology_Efficacy_Plots.Rmd (hunk 7 above)
Comment (margoal1) [reply Rmarkdown/Oncology_Efficacy_Plots.Rmd]: similar to DOSE_label_low2high and DOSE_label_high2low?

Comment (margoal1) [reply Rmarkdown/Oncology_Efficacy_Plots.Rmd]: Maybe name them MONO_DOSE_label_low2high and COMBO_DOSE_label_low2high?

File: Rmarkdown/Oncology_Efficacy_Plots.Rmd [hunk 9]
```diff
@@ -41,17 +38,46 @@ xgx_theme_set()
 
 
 ```{r, warning=FALSE, message=FALSE}
-
 my.data <- read.csv("../Data/Oncology_Efficacy_Data.csv")
```
Comment (margoal1) [inline Rmarkdown/Oncology_Efficacy_Plots.Rmd]: what name would better describe this data, onc_data? recist_data?

File: Rmarkdown/Oncology_Efficacy_Plots.Rmd [hunk 10]
```diff
@@ -41,17 +38,46 @@ xgx_theme_set()
...
+
+dose_record = dose_record %>%
+  mutate(IDSHORT      =  IDSHORT,         #ID column
+         DOSE         =  DOSE,            #Dose amount
+         TIME         = TIME,             #Dosing time
+         COMB         = COMB,             #Single agent or combo
+         )
+
+# make single agent and combo datsets
 my.data.monotherapy = my.data %>% filter(COMB=="Single")
```
Comment (margoal1) [inline Rmarkdown/Oncology_Efficacy_Plots.Rmd]: recist_data_mono?

File: Rmarkdown/Oncology_Efficacy_Plots.Rmd [hunk 11]
```diff
@@ -41,17 +38,46 @@ xgx_theme_set()
...
 my.data.combo = my.data %>% filter(COMB=="Combo")
```
Comment (margoal1) [inline Rmarkdown/Oncology_Efficacy_Plots.Rmd]: recist_data_combo?

File: Rmarkdown/Oncology_Efficacy_Plots.Rmd [hunk 12]
```diff
@@ -77,6 +103,31 @@ dose_record <- dose_record %>%
...
+
+#units and labels
+time_units_dataset     = "hours"
+time_units_plot        = "days"
+trtact_label           = "Dose"
+dose_label             = "Dose (mg)"
+time_label             = "Time(Days)"
+timemonth_label        = "Time (months)"
+sld_label              = "Percent Change in\nSum of Longest Diameters"
+sadose_label           = "ABC123 Dose (mg)"
```
Comment (margoal1) [inline Rmarkdown/Oncology_Efficacy_Plots.Rmd]: mono_dose_label?

File: Rmarkdown/Oncology_Efficacy_Plots.Rmd [hunk 13]
```diff
@@ -77,6 +103,31 @@ dose_record <- dose_record %>%
...
+combodose_label        = "DEF456 Dose (mg)"
```
Comment (margoal1) [inline Rmarkdown/Oncology_Efficacy_Plots.Rmd]: combo_dose_label

Comment (margoal1) [reply Rmarkdown/Oncology_Efficacy_Plots.Rmd]: it's easier to read if you add "_" between different words

File: Rmarkdown/Oncology_Efficacy_Plots.Rmd [hunk 14]
```diff
@@ -77,6 +103,31 @@ dose_record <- dose_record %>%
...
+bpchg_label            = "Best Percent Change \n in Tumor Size (%)"
+saauc_label            = "ABC123 AUC0-24 (ng.h/ml)"
+saaveauc_label         = "Average ABC123 dose (mg) \n up to the first assessment"
```
Comment (margoal1) [inline Rmarkdown/Oncology_Efficacy_Plots.Rmd]: "avg" is a more common abbreviation for average, my first thought when I see "ave" is "avenue"

Comment (margoal1) [reply Rmarkdown/Oncology_Efficacy_Plots.Rmd]: mono_avg_auc_label

File: Rmarkdown/Oncology_Efficacy_Plots.Rmd [hunk 15]
```diff
@@ -77,6 +103,31 @@ dose_record <- dose_record %>%
...
+bpchgfirstassess_label = " Percent Change in Tumor Size \n up to the first assessment"
```
Comment (margoal1) [inline Rmarkdown/Oncology_Efficacy_Plots.Rmd]: first_pchg_label

Comment (faribas) [reply Rmarkdown/Oncology_Efficacy_Plots.Rmd]: done!

File: Rmarkdown/Oncology_Efficacy_Plots.Rmd (hunk 14 above)
Comment (faribas) [reply Rmarkdown/Oncology_Efficacy_Plots.Rmd]: done!

File: Rmarkdown/Oncology_Efficacy_Plots.Rmd (hunk 13 above)
Comment (faribas) [reply Rmarkdown/Oncology_Efficacy_Plots.Rmd]: done!

File: Rmarkdown/Oncology_Efficacy_Plots.Rmd (hunk 12 above)
Comment (faribas) [reply Rmarkdown/Oncology_Efficacy_Plots.Rmd]: done!

File: Rmarkdown/Oncology_Efficacy_Plots.Rmd (hunk 8 above)
Comment (faribas) [reply Rmarkdown/Oncology_Efficacy_Plots.Rmd]: done!

File: Rmarkdown/Oncology_Efficacy_Plots.Rmd (hunk 7 above)
Comment (faribas) [reply Rmarkdown/Oncology_Efficacy_Plots.Rmd]: done!

File: Rmarkdown/Oncology_Efficacy_Plots.Rmd (hunk 10 above)
Comment (faribas) [reply Rmarkdown/Oncology_Efficacy_Plots.Rmd]: done!

**Pull request merged**